src/mscorlib/src/System/Environment.cs

```diff
@@ -379,9 +379,9 @@ public static String CurrentDirectory
             [System.Security.SecuritySafeCritical]  // auto-generated
 #endif
             get {
-                StringBuilder sb = new StringBuilder(Path.MAX_PATH);
-                int r = Win32Native.GetSystemDirectory(sb, Path.MAX_PATH);
-                Contract.Assert(r < Path.MAX_PATH, "r < Path.MAX_PATH");
+                StringBuilder sb = new StringBuilder(Path.MaxPath);
```

Comment:
Not for this PR, but any idea why we're using StringBuilderCache in some places for getting paths (e.g. Directory.GetCurrentDirectory) but not in others (e.g. here in SystemDirectory)?